src/AwkwardArray.jl

```diff
@@ -9,5 +9,9 @@ include("./tables.jl")
 include("./AwkwardPythonCallExt.jl")
 using .AwkwardPythonCallExt: convert
 
+Base.eltype(::RecordArray{FIELDS,CONTENTS,BEHAVIOR}) where {FIELDS,CONTENTS,BEHAVIOR} = Record{FIELDS,CONTENTS,BEHAVIOR}
+Base.eltype(::Record{FIELDS,CONTENTS,BEHAVIOR}) where {FIELDS,CONTENTS,BEHAVIOR} = CONTENTS
```

Comment:
Does a `Record` (a scalar object, not an array) even have an `eltype`?

Reply:
I think it depends on if you model the `Record` as an indexable thing. For me at least, I think event-based table semantically as a row table, so the `eltype` is whatever the type an "event" would give you

Reply:
I'm not sure if it makes sense to separate `FIELDS` and `CONTENTS`, but perhaps there is a usecase for it?

```julia
julia> typeof(layout[3])
AwkwardArray.Record{(:a, :b), Tuple{AwkwardArray.PrimitiveArray{Int64, Vector{Int64}, :default}, AwkwardArray.ListOffsetArray{Vector{Int64}, AwkwardArray.PrimitiveArray{Float64, Vector{Float64}, :default}, :default}}, :default}

julia> r = layout[3]
{a: 3, b: [4.4, 5.5]}

julia> eltype(r)
Tuple{AwkwardArray.PrimitiveArray{Int64, Vector{Int64}, :default}, AwkwardArray.ListOffsetArray{Vector{Int64}, AwkwardArray.PrimitiveArray{Float64, Vector{Float64}, :default}, :default}}
```